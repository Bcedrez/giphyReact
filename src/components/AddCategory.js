import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCatergory = ({setCategorias}) => {


    const [valorInput, SetValorInput] = useState('')

    const handleValorInput = (e) => {
        SetValorInput(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(valorInput.trim().length > 1){
            setCategorias(categorias =>[valorInput,...categorias])
            SetValorInput('')
        }
        
        
    }

    return (

      
        <form onSubmit={handleSubmit}>
            <input
            placeholder="Buscar Gif"
            type="text"
            value={valorInput}
            onChange={handleValorInput}
            />
        </form>
    );
}

AddCatergory.propTypes = {
    setCategorias : PropTypes.func.isRequired
}