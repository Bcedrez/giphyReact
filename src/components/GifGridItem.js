import React from "react";
import PropTypes from 'prop-types';


const GifGridItem = ({id,titulo,url})=>{
    // console.log({id,titulo,url})
    return (
        <div className="card animate__animated animate__zoomIn">
           <img src={url} alt={titulo} />
           <p>{titulo}</p>
        </div>
    );
}

GifGridItem.propTypes = {
    titulo: PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
}


export default GifGridItem