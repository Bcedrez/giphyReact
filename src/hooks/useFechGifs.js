import { useEffect, useState } from "react"
import { getGift } from "../helpers/getGifs";

export const useFetchGifs = (categoria)=>{

    const [state,setState] = useState({
        data:[],
        loading:true
    })

    useEffect(()=>{

        getGift(categoria)
            .then(imgs => {


                setTimeout(()=>{
                    setState({
                        data:imgs,
                        loading:false
                    })
                    
                },3000)
                // setState({
                //     data:imgs,
                //     loading:false
                // })
            })
        
    },[categoria]);

   
    return state
}