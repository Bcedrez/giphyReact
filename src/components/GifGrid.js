import React  from "react";

import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks/useFechGifs";
import  GifGridItem  from "./GifGridItem";



export const GifGrid = ({categoria})=>{
    
    const {data:imagenes,loading} = useFetchGifs(categoria);   

    return (
        <>
        <h3 className="animate__animated animate__rotateInDownLeft">{categoria}</h3>

        {loading && <p>loading...</p>}
        
        <div className="card-grid">  
            
                {
                    imagenes.map(imagen =>{
                        return <GifGridItem 
                                    key={imagen.id} 
                                    {...imagen}
                                   
                                     />
                    })
                }     
            
        </div>
       </> 
    )
}

GifGrid.propTypes = {
    categoria : PropTypes.string.isRequired
}