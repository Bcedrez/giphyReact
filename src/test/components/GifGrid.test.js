import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFechGifs';
jest.mock('../../hooks/useFechGifs');


describe('Pruebas en componente <GifGrid />',()=>{


   
    const categoria = 'Prueba';
    // const wrapper = shallow(<GifGrid categoria={categoria} />);

    beforeEach(()=>{
        jest.clearAllMocks();
        
    });

    test('Ver que se renderize bien',()=>{   

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });

        const wrapper = shallow(<GifGrid categoria={categoria} />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imagenes por useFetchGihs',()=>{

        const gifs = [
            {id:'ABC',url:'https://localhost/algo.jpg',titulo:'Un titulo'},
            {id:'BCD',url:'https://localhost/algomas.jpg',titulo:'Un titulo 2'}]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid categoria={categoria} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });

});