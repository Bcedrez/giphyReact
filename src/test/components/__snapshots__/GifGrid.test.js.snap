// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes por useFetchGihs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rotateInDownLeft"
  >
    Prueba
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      titulo="Un titulo"
      url="https://localhost/algo.jpg"
    />
    <GifGridItem
      id="BCD"
      key="BCD"
      titulo="Un titulo 2"
      url="https://localhost/algomas.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente <GifGrid /> Ver que se renderize bien 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rotateInDownLeft"
  >
    Prueba
  </h3>
  <p>
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
